File: dfeToken.cs
using System;
using System.Security.Cryptography;
using System.Text;
using Newtonsoft.Json;

public class SecureToken
{
    public static string CreateSecureTokenDFe(string password, string uuid)
    {
        password = System.Text.RegularExpressions.Regex.Replace(password, "[^0-9]", "");
        using (SHA256 sha256 = SHA256.Create())
        {
            string keySource = password + ":" + uuid;
            byte[] key = sha256.ComputeHash(Encoding.UTF8.GetBytes(keySource));
            using (Aes aes = Aes.Create())
            {
                aes.Key = key;
                aes.GenerateIV();
                byte[] iv = aes.IV;
                aes.Mode = CipherMode.CBC;
                aes.Padding = PaddingMode.PKCS7;
                using (ICryptoTransform encryptor = aes.CreateEncryptor(aes.Key, iv))
                {
                    string timeString = DateTimeOffset.UtcNow.ToUnixTimeSeconds().ToString();
                    byte[] timeBytes = Encoding.UTF8.GetBytes(timeString);
                    byte[] encryptedData = encryptor.TransformFinalBlock(timeBytes, 0, timeBytes.Length);
                    var tokenData = new
                    {
                        data = Convert.ToBase64String(encryptedData),
                        iv = Convert.ToBase64String(iv)
                    };
                    string tokenJson = JsonConvert.SerializeObject(tokenData);
                    return Uri.EscapeDataString(Convert.ToBase64String(Encoding.UTF8.GetBytes(tokenJson)));
                }
            }
        }
    }

    public static void Main()
    {
        string password = "00.000.000/0000-00"; // CPF/CNPJ do tomador
        string uuid = "00000000-0000-0000-0000-000000000000"; // UUID da nota fiscal
        string key = "00000000000000000000000000000000000000000000"; // Chave da nota fiscal
        string token = CreateSecureTokenDFe(password, uuid);
        string url = $"https://nfe.webmaniabr.com/danfe/{key}/?token={token}";

        Console.WriteLine(url);
    }
}

File: dfeToken.js
async function createSecureTokenDFe(password, uuid) {
  const numericPassword = String(password).replace(/[^0-9]/g, '');
  const dataToHash = `${numericPassword}:${uuid}`;

  const encoder = new TextEncoder();
  const hash = await crypto.subtle.digest('SHA-256', encoder.encode(dataToHash));
  const key = await crypto.subtle.importKey('raw', hash, { name: 'AES-CBC' }, false, ['encrypt']);

  const iv = crypto.getRandomValues(new Uint8Array(16));
  const nowSeconds = Math.floor(Date.now() / 1000).toString();

  const encrypted = await crypto.subtle.encrypt(
    { name: 'AES-CBC', iv },
    key,
    encoder.encode(nowSeconds)
  );

  const tokenData = JSON.stringify({
    data: toBase64(encrypted),
    iv: toBase64(iv),
  });

  return encodeURIComponent(btoa(tokenData));
}

function toBase64(input) {
  const bytes = input instanceof Uint8Array ? input : new Uint8Array(input);
  let binary = '';
  for (let i = 0; i < bytes.length; i++) binary += String.fromCharCode(bytes[i]);
  return btoa(binary);
}

(async () => {
  const password = '00.000.000/0000-00'; // CPF/CNPJ do tomador
  const uuid = '00000000-0000-0000-0000-000000000000'; // UUID da nota fiscal
  const chaveDaNota = '00000000000000000000000000000000000000000000'; // Chave da nota fiscal

  const token = await createSecureTokenDFe(password, uuid);
  const url = `https://nfe.webmaniabr.com/danfe/${chaveDaNota}/?token=${token}`;
})();
